### ISSUE TITLE
[fanfic.tenhawkpresents.ink] TypeError: ord() expected string of length 1, but int found

### ISSUE THREAD

**Issue by nsapa** [JimmXinu/FanFicFare#483]
Good afternoon,

FanFicFare fail on [chapter 15](http://fanfic.tenhawkpresents.ink/viewstory.php?sid=184&textsize=0&chapter=15) of this [story](http://fanfic.tenhawkpresents.ink/viewstory.php?sid=184). 

From the log, it seem the encoding is wrong and it fail stripping non-ASCII.

Relevant log:

```
FFF: INFO: 2020-04-12 16:47:19,408: calibre_plugins.fanficfare_plugin.jobs(53): FanFicFare v3.18.0
FFF: INFO: 2020-04-12 16:47:19,408: calibre_plugins.fanficfare_plugin.jobs(57): Adding jobs for URLs:
FFF: INFO: 2020-04-12 16:47:19,408: calibre_plugins.fanficfare_plugin.jobs(59): http://fanfic.tenhawkpresents.ink/viewstory.php?sid=184
FFF: INFO: 2020-04-12 16:47:54,787: calibre_plugins.fanficfare_plugin.jobs(95): Logfile for book ID None (Futures Insight 01: Futures Insight)
FFF: INFO: 2020-04-12 16:47:54,788: calibre_plugins.fanficfare_plugin.jobs(96): url:(http://fanfic.tenhawkpresents.ink/viewstory.php?sid=184) id:(None)
FFF: DEBUG: 2020-04-12 16:47:20,937: calibre_plugins.fanficfare_plugin.fanficfare.adapters.adapter_tenhawkpresents(112): URL: http://fanfic.tenhawkpresents.ink/viewstory.php?sid=184&index=1&ageconsent=ok&warning=3
FFF: DEBUG: 2020-04-12 16:47:20,937: calibre_plugins.fanficfare_plugin.fanficfare.configurable(1118): #####################################
pagecache(GET) HIT: http://fanfic.tenhawkpresents.ink/viewstory.php?sid=184&index=1&ageconsent=ok&warning=3
FFF: DEBUG: 2020-04-12 16:47:20,937: calibre_plugins.fanficfare_plugin.fanficfare.configurable(991): Encoding:Windows-1252
FFF: INFO: 2020-04-12 16:47:21,107: calibre_plugins.fanficfare_plugin.jobs(246): write to /tmp/calibre_4.12.0_tmp_ospn41n4/fanficfare_fnihkiv_/Futures Insight 01_ Futures Insight-lucifael-kzeeb6nd.epub
FFF: INFO: 2020-04-12 16:47:21,108: calibre_plugins.fanficfare_plugin.fanficfare.writers.base_writer(183): Save directly to file: /tmp/calibre_4.12.0_tmp_ospn41n4/fanficfare_fnihkiv_/Futures Insight 01_ Futures Insight-lucifael-kzeeb6nd.epub
...
FFF: DEBUG: 2020-04-12 16:47:52,373: calibre_plugins.fanficfare_plugin.fanficfare.adapters.adapter_tenhawkpresents(239): Getting chapter text from: http://fanfic.tenhawkpresents.ink/viewstory.php?sid=184&chapter=15&ageconsent=ok&warning=3
FFF: DEBUG: 2020-04-12 16:47:52,373: calibre_plugins.fanficfare_plugin.fanficfare.configurable(1128): #####################################
pagecache(GET) MISS: http://fanfic.tenhawkpresents.ink/viewstory.php?sid=184&chapter=15&ageconsent=ok&warning=3
FFF: DEBUG: 2020-04-12 16:47:54,522: calibre_plugins.fanficfare_plugin.fanficfare.configurable(991): Encoding:Windows-1252
FFF: DEBUG: 2020-04-12 16:47:54,522: calibre_plugins.fanficfare_plugin.fanficfare.configurable(1012): code failed:Windows-1252
FFF: DEBUG: 2020-04-12 16:47:54,522: calibre_plugins.fanficfare_plugin.fanficfare.configurable(1013): 'charmap' codec can't decode byte 0x81 in position 37166: character maps to <undefined>
FFF: DEBUG: 2020-04-12 16:47:54,522: calibre_plugins.fanficfare_plugin.fanficfare.configurable(991): Encoding:utf8
FFF: DEBUG: 2020-04-12 16:47:54,522: calibre_plugins.fanficfare_plugin.fanficfare.configurable(1012): code failed:utf8
FFF: DEBUG: 2020-04-12 16:47:54,522: calibre_plugins.fanficfare_plugin.fanficfare.configurable(1013): 'utf-8' codec can't decode byte 0x92 in position 6456: invalid start byte
FFF: INFO: 2020-04-12 16:47:54,522: calibre_plugins.fanficfare_plugin.fanficfare.configurable(1015): Could not decode story, tried:['Windows-1252', 'utf8'] Stripping non-ASCII.
FFF: DEBUG: 2020-04-12 16:47:54,522: calibre_plugins.fanficfare_plugin.fanficfare.configurable(1218): Caught an exception reading URL: http://fanfic.tenhawkpresents.ink/viewstory.php?sid=184&chapter=15&ageconsent=ok&warning=3 sleeptime(0) Exception ord() expected string of length 1, but int found.
FFF: INFO: 2020-04-12 16:47:54,523: calibre_plugins.fanficfare_plugin.jobs(346): Exception: {'title': 'Futures Insight 01: Futures Insight', 'author_sort': ['lucifael'], 'author': ['lucifael'], 'comments': "<p>Girls, Technology, The TARDIS, Magic and Music. Harry's starting a new life after his old one get's kinda boring.</p><br/>", 'good': False, 'status': 'Erreur', 'showerror': True, 'calibre_id': None, 'begin': None, 'end': None, 'comment': 'ord() expected string of length 1, but int found', 'url': 'http://fanfic.tenhawkpresents.ink/viewstory.php?sid=184', 'site': 'fanfic.tenhawkpresents.ink', 'series': '', 'added': False, 'pubdate': datetime.datetime(2008, 11, 20, 0, 0, tzinfo=SafeLocalTimeZone()), 'publisher': 'fanfic.tenhawkpresents.ink', 'uniqueurl': 'http://fanfic.tenhawkpresents.ink/viewstory.php?sid=184[None-None]', 'listorder': 0, 'collision': 'Update EPUB if New Chapters', 'is_adult': False, 'username': 'NoneGiven', 'password': '', 'icon': 'dialog_error.png', 'all_metadata': {'category': 'Harry Potter, Multi-Cross', 'genre': '', 'characters': 'Harry Potter', 'ships': '', 'warnings': '', 'extratags': 'FanFiction', 'author': 'lucifael', 'authorId': '195', 'authorUrl': 'http://fanfic.tenhawkpresents.ink/viewuser.php?uid=195', 'lastupdate': 'Last Update Year/Month: 2009/01, Last Update: 2009/01/16', 'series': '', 'seriesUrl': '', 'language': '', 'status': 'Completed', 'datePublished': '2008-11-20', 'dateUpdated': '2009-01-16', 'dateCreated': '2020-04-12 16:47:16', 'rating': 'M', 'numChapters': '26', 'numWords': '164,920', 'words_added': '', 'site': 'fanfic.tenhawkpresents.ink', 'publisher': 'fanfic.tenhawkpresents.ink', 'storyId': '184', 'title': 'Futures Insight 01: Futures Insight', 'titleHTML': "<a class='titlelink' href='http://fanfic.tenhawkpresents.ink/viewstory.php?sid=184'>Futures Insight 01: Futures Insight</a>", 'storyUrl': 'http://fanfic.tenhawkpresents.ink/viewstory.php?sid=184', 'sectionUrl': 'http://fanfic.tenhawkpresents.ink/viewstory.php?sid=184', 'description': "<p>Girls, Technology, The TARDIS, Magic and Music. Harry's starting a new life after his old one get's kinda boring.</p><br/>", 'formatname': '', 'formatext': '', 'siteabbrev': 'thpi', 'version': 'FanFicFare v3.18.0', 'authorHTML': "<a class='authorlink' href='http://fanfic.tenhawkpresents.ink/viewuser.php?uid=195'>lucifael</a>", 'seriesHTML': '', 'langcode': '', 'output_css': '', 'cover_image': ''}, 'tags': ['Multi-Cross', 'Completed', 'FanFiction', 'Harry Potter'], 'updatedate': datetime.datetime(2009, 1, 16, 0, 0, tzinfo=SafeLocalTimeZone()), 'timestamp': datetime.datetime(2020, 4, 12, 16, 47, 20, 931478, tzinfo=SafeLocalTimeZone()), 'calibre_columns': {}, 'outfile': '/tmp/calibre_4.12.0_tmp_ospn41n4/fanficfare_fnihkiv_/Futures Insight 01_ Futures Insight-lucifael-kzeeb6nd.epub'}:ord() expected string of length 1, but int found
Traceback (most recent call last):
  File "calibre_plugins.fanficfare_plugin.jobs", line 248, in do_download_for_worker
  File "calibre_plugins.fanficfare_plugin.fanficfare.writers.base_writer", line 202, in writeStory
  File "calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter", line 237, in getStory
  File "calibre_plugins.fanficfare_plugin.fanficfare.adapters.base_adapter", line 389, in getChapterTextNum
  File "calibre_plugins.fanficfare_plugin.fanficfare.adapters.adapter_tenhawkpresents", line 241, in getChapterText
  File "calibre_plugins.fanficfare_plugin.fanficfare.configurable", line 1333, in _fetchUrl
  File "calibre_plugins.fanficfare_plugin.fanficfare.configurable", line 1322, in _fetchUrlOpened
  File "calibre_plugins.fanficfare_plugin.fanficfare.configurable", line 1204, in _fetchUrlOpened
  File "calibre_plugins.fanficfare_plugin.fanficfare.configurable", line 1016, in _decode
  File "calibre_plugins.fanficfare_plugin.fanficfare.configurable", line 1016, in <listcomp>
TypeError: ord() expected string of length 1, but int found
```

I am running FanFicFare v3.18.0 via Calibre 4.12 on Fedora 31 where python is Python 3.7.6.

Thank you for this software.

Best regards,
NS

**Comment by JimmXinu** [comment]
That's a py2/3 issue.  It should be fixed in 968974706371e63e09defa4b008b529528723684.

You can try it from the test version with:
`pip install --extra-index-url https://testpypi.python.org/pypi --upgrade FanFicFare`

(Thanks for giving a concise yet complete problem report.)

**Comment by nsapa** [comment]
I confirm that version 3.18.5 fix this issue.

Thanks you and have a nice day!
